File: api/src/main/java/com/wbrawner/budgetserver/config/SessionAuthenticationToken.java
package com.wbrawner.budgetserver.config;

import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.GrantedAuthority;

import java.util.Collection;

public class SessionAuthenticationToken extends UsernamePasswordAuthenticationToken {
    /**
     * Creates a token with the supplied array of authorities.
     *
     * @param authorities the collection of <tt>GrantedAuthority</tt>s for the principal
     *                    represented by this authentication object.
     * @param credentials
     * @param principal
     */
    public SessionAuthenticationToken(Object principal, Object credentials, Collection<? extends GrantedAuthority> authorities) {
        super(principal, credentials, authorities);
    }
}

File: app/src/main/kotlin/com/wbrawner/twigs/server/config/SessionAuthenticationToken.kt
package com.wbrawner.twigs.server.config

import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.security.core.GrantedAuthority

/**
 * Creates a token with the supplied array of authorities.
 *
 * @param authorities the collection of <tt>GrantedAuthority</tt>s for the principal
 * represented by this authentication object.
 * @param credentials
 * @param principal
 */
class SessionAuthenticationToken(
    principal: Any?,
    credentials: Any?,
    authorities: Collection<GrantedAuthority>
) : UsernamePasswordAuthenticationToken(principal, credentials, authorities)